src/OrchardCore/OrchardCore.Data.YesSql/SqliteHelper.cs

```diff
@@ -9,14 +9,23 @@ public static class SqliteHelper
     public static string GetConnectionString(SqliteOptions sqliteOptions, ShellOptions shellOptions, string shellName) =>
         GetConnectionString(sqliteOptions, GetDatabaseFolder(shellOptions, shellName));
 
-    public static string GetConnectionString(SqliteOptions sqliteOptions, string databaseFolder) =>
-        new SqliteConnectionStringBuilder
+    public static string GetConnectionString(SqliteOptions sqliteOptions, string databaseFolder)
+    {
+        var connectionStringBuilder = new SqliteConnectionStringBuilder
         {
-            DataSource = Path.Combine(databaseFolder, sqliteOptions.DatabaseName),
             Cache = SqliteCacheMode.Shared,
             Pooling = sqliteOptions.UseConnectionPooling
+        };
+        var dataSource = Path.Combine(databaseFolder, sqliteOptions.DatabaseName);
+        if (!File.Exists(dataSource))
```

Comment:
I don't this this is a good idea to do IO operation for this. I still like the approach I recommended in the issue better

Reply:
This runs on the first request to the tenant if I'm not wrong

Reply:
You want to store the database name in the shellsetting config file instead. The issue will still remain if people are not changing it.

Reply:
If value is null from shellsettings then return "yessql.db". That could work. But you are adding a new setting that was not necessary before. Is the intent to add that config to the shellsettings to make this dynamic too?

Reply:
@Skrypt is saying the same recommendation I added here https://github.com/OrchardCMS/OrchardCore/issues/15154

@hishamco I can submit alternative PR if you like just so you can compare the two.

Reply:
> You want to store the database name in the shellsetting config file instead. The issue will still remain if people are not changing it.

I understand the opposite :)

Reply:
@Skrypt are you with or without adding a new setting?

Reply:
It makes sense if the intent is to make this name configurable. I thought that the intent was to only change its default name. But yeah, to be honest I don't know what is the added value of making this name configurable from the shellSettings json file. Why would someone want to rename that file is the question?

It has always been a fixed value since the beginning but I'm not against the idea. Though, it will add complexity and it needs to be documented and it needs to throw if the DB is not found when someone puts in a random name in these shellshettings.

Reply:
We might you the tenant's name while it's safe, the idea at the beginning to rid-off the `yessql`

Reply:
@Skrypt I remember the configuration was done mainly for backward compatibility, so you can `yessql.db` from within the configuration

For that this PR might not necessary

Reply:
What I don't like about making the Databasename configurable from the appsettings.json file is that it adds a failure point that was not there before. If we do this then at least we need to throw/log ...

Reply:
I don't think we should care. If someone decided to screw up their appsettings file then their app won't work. Same as changing the tenant name or anything else like tablePrefix. I think we should use appsettings and assume that the file matches the name if one found otherwise yessql. I submitted a PR that fixes this issue #15157